tools/shared.py

```diff
@@ -127,6 +127,17 @@ def run_process(cmd, check=True, input=None, *args, **kw):
   sys.stderr.flush()
   kw.setdefault('text', True)
   kw.setdefault('encoding', 'utf-8')
+  # Handle calls to the Python scripts by going through the python executable
+  if cmd and cmd[0].endswith('.py'):
+    cmd = [sys.executable, '-E'] + cmd
+    # $PYTHON -E will not ignore _PYTHON_SYSCONFIGDATA_NAME, which is an internal
+    # of cpython used in cross compilation via setup.py.
+    environ = kw.get('env')
+    if environ is None:
+      environ = os.environ.copy()
+    if '_PYTHON_SYSCONFIGDATA_NAME' in environ:
+      del environ['_PYTHON_SYSCONFIGDATA_NAME']
+      kw['env'] = environ
```

Comment:
I would prefer not to modify this function, but instead replace `bat_suffix` with something like `make_py_cmd`

Reply:
The problem with such a function is that it doesn't handle the removal of `_PYTHON_SYSCONFIGDATA_NAME` from the environment.

Reply:
Ah, but assuming that a .bat file was used to launch to initial process I think we can maybe assume that `_PYTHON_SYSCONFIGDATA_NAME` was cleared already?

Reply:
Lets just assume `_PYTHON_SYSCONFIGDATA_NAME` was removed by the outer batch file.  The removal of `_PYTHON_SYSCONFIGDATA_NAME` is a pretty obscure/strange thing we need to do IIRC its only needed when building CPython itself.

Reply:
Another issue with a `make_py_cmd` approach is that we no longer have a single string, but a list of strings (because we need to add `[sys.executable, '-E']` in front of it), which means some tool constants at the end of `shared.py` would be plain string while others lists, causing confusion.